packages/web3-errors/src/errors/core_errors.ts

```diff
@@ -0,0 +1,41 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* eslint-disable max-classes-per-file */
+
+import { Web3Error } from '../web3_error_base';
+import { ERR_CORE_HARDFORK_MISMATCH } from '../error_codes';
+
+export class ConfigHardforkMismatchError extends Web3Error {
+	public code = ERR_CORE_HARDFORK_MISMATCH;
+
+	public constructor(defaultHardfork: string, commonHardFork: string) {
+		super(
+			`Web3Config hardfork doesnt match in defaultHardfork ${defaultHardfork} and common.hardfork ${commonHardFork}`,
+		);
+	}
+}
+
+export class ConfigChainMismatchError extends Web3Error {
+	public code = ERR_CORE_HARDFORK_MISMATCH;
```

Comment:
I think you meant this class to be about chain chain here.

Reply:
These are two different errors although they look the same, they occur at different packages. `ERR_TX_CHAIN_MISMATCH` is for validation within a transaction signing in web3-eth and `ERR_CORE_HARDFORK_MISMATCH` is for web3config in web3-core